File: src/main/webapp/WEB-INF/static/core/service/UsuarioPerfilCampeonatoService.js
/**
 * UsuarioPerfilCampeonatoService
 */
app.factory('UsuarioPerfilCampeonatoService', ['$http', function($http) {
	
	var _serverURL = function(url) {
		return SERVER_APP + '/usuarioPerfilCampeonato' + url;
	};
	
	/**
	 * pega UsuarioPerfilCampeonato por usuario e campeonato
	 */
	var _getByUsuarioAndCampeonato = function(usuario, campeonato) {
		return $http.get(_serverURL('/get/by/usuario/'+usuario.id+'/and/campeonato/'+campeonato.id));
	};
	
	/**
	 * pega lista por usuario
	 */
	var _getListByUsuario = function(usuario) {
		return $http.get(_serverURL('/get/list/by/usuario/'+usuario.id));
	};
	
	return {
		getByUsuarioAndCampeonato : _getByUsuarioAndCampeonato,
		
		getListByUsuario : _getListByUsuario
	};
	
}]);

File: src/main/java/com/diegolirio/jcampeonato/service/UsuarioPerfilCampeonatoService.java
package com.diegolirio.jcampeonato.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

import com.diegolirio.jcampeonato.dao.UsuarioPerfilCampeonatoDao;
import com.diegolirio.jcampeonato.model.Campeonato;
import com.diegolirio.jcampeonato.model.Usuario;
import com.diegolirio.jcampeonato.model.UsuarioPerfilCampeonato;

@Service("/usuarioPerfilCampeonatoService")
public class UsuarioPerfilCampeonatoService extends AbstractGenericService<UsuarioPerfilCampeonato> {

	@Autowired @Qualifier("usuarioPerfilCampeonatoDao")
	private UsuarioPerfilCampeonatoDao usuarioPerfilCampeonatoDao;

	public UsuarioPerfilCampeonato getByUsuarioAndCampeonato(Usuario usuario, Campeonato campeonato) {
		return this.usuarioPerfilCampeonatoDao.getByUsuarioAndCampeonato(usuario, campeonato);
	}

	public List<UsuarioPerfilCampeonato> getListByUsuario(Usuario usuario) {
		return this.usuarioPerfilCampeonatoDao.getListByUsuario(usuario); 
	}

}
